src/OrbitGl/App.h

```diff
@@ -511,6 +511,13 @@ class OrbitApp final : public DataViewFactory,
   [[nodiscard]] const orbit_statistics::BinomialConfidenceIntervalEstimator&
   GetConfidenceIntervalEstimator() const override;
 
+  void OnHistogramSelectionRangeChange() { RequestUpdatePrimitives(); }
+
```

Comment:
I wouldn't call this function `OnHistogramSelectionRangeChange`. Ideally this function doesn't know that it's an event listener from somewhere else. (Decoupling again, I'm sorry. :smile: )

Considering my comment below, I would treat it like a setter and call it `SetHistogramSelectionRange`. That means the histogram selection range becomes a simple property of OrbitApp. No dependencies to any other components. And the common parent of the histogram widget and OrbitApp (which is OrbitMainWindow) has the responsibility to keep those two in sync (by calling a signal-slot connection)